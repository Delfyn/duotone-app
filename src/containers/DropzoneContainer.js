/* @flow */

import { connect } from 'react-redux';
import { setInvalidFile, addFile } from '../actions';
import Dropzone from './Dropzone';

const mapStateToProps = state => ({
  ...state,
});

const mapDispatchToProps = dispatch => ({
  invalidFile: (primary, secondary) => {
    dispatch(setInvalidFile(primary, secondary));
  },
  addFile: (preview, size, type, name) => {
    dispatch(addFile(preview, size, type, name));
  },
});

const DropzoneContainer = connect(
  mapStateToProps,
  mapDispatchToProps,
)(Dropzone);

export default DropzoneContainer;
