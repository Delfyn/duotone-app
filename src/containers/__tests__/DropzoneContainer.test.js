import React from 'react';
import renderer from 'react-test-renderer';
import DropzoneContainer from '../DropzoneContainer';

test('DropzoneContainer renders correctly', () => {
  const state = {};

  const subscribe = jest.fn();
  const dispatch = jest.fn();
  const getState = jest.fn(() => state);

  const store = { subscribe, dispatch, getState };
  const props = {};

  const component = renderer.create(
    <DropzoneContainer store={store} {...props} />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  expect(subscribe).toBeCalled();
  expect(getState).toBeCalled();
});
