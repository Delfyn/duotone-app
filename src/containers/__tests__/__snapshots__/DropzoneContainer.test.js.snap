// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropzoneContainer renders correctly 1`] = `
<div
  className=""
  onClick={[Function]}
  onDragEnter={[Function]}
  onDragLeave={[Function]}
  onDragOver={[Function]}
  onDragStart={[Function]}
  onDrop={[Function]}
  style={Object {}}
>
  <div
    className="drop-area z3"
  />
  <input
    accept="image/*"
    multiple={false}
    onChange={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
    type="file"
  />
</div>
`;
